libraries/AP_NavEKF3/AP_NavEKF3_MagFusion.cpp

```diff
@@ -328,11 +328,15 @@ void NavEKF3_core::SelectMagFusion()
         }
     }
 
-    // If we have no magnetometer, fuse in a synthetic heading measurement at 7Hz to prevent the filter covariances
-    // from becoming badly conditioned. For planes we only do this on-ground because they can align the yaw from GPS when
-    // airborne. For other platform types we do this all the time.
-    if (!use_compass()) {
-        if ((onGround || !assume_zero_sideslip()) && (imuSampleTime_ms - lastSynthYawTime_ms > 140)) {
+    // we require continuous direct yaw observations because we are not a fly forward vehicle in flight
+    bool directYawObsRequired = !(assume_zero_sideslip() && !onGround);
+
+    // we have not received the data required to directly observe yaw for more than 15 seconds
+    bool directYawObsTimeout = ((imuSampleTime_ms - yawAngDataDelayed.time_ms) > 15000 && frontend->_magCal == 5)
```

Comment:
If _magCal is 5, we are never reach here, because 
`

    // Handle case where we are using an external yaw sensor instead of a magnetomer
    if ((frontend->_magCal == 5)) {
        if (storedYawAng.recall(yawAngDataDelayed,imuDataDelayed.time_ms)) {
            if (tiltAlignComplete && !yawAlignComplete) {
                alignYawAngle();
            } else if (tiltAlignComplete && yawAlignComplete) {
                fuseEulerYaw(false, true);
            } else {
                fuseEulerYaw(true, true);
            }
        }
        return;
    }
`

Reply:
Will fix it

Reply:
en.The GCS did not display the message.

Reply:
en.The GCS did not display the message  --"EKF3 IMU%u yaw aligned"